ext/OptimizationEnzymeExt.jl

```diff
@@ -13,30 +13,36 @@ function Optimization.instantiate_function(f::OptimizationFunction{true}, x,
 
     if f.grad === nothing
         function grad(res, θ, args...)
-            Enzyme.gradient!(Enzyme.Reverse, res, (θ) -> f.f(θ, p, args...), θ)
+            dθ = zero(res)
```

Comment:
you could just set res to zero here, then pass it in as dtheta, instead of making the temporary.